mkosi

```diff
@@ -56,6 +57,14 @@ if sys.version_info < (3, 6):
     sys.exit("Sorry, we need at least Python 3.6.")
 
 
+if TYPE_CHECKING:
```

Comment:
I didn't read the other PRs yet so this might have been clarified already but a comment in the source code explaining why we do this would be great.

Reply:
Do you mean a specific comment for this block or why typing in general?

Reply:
This block, why typing should hopefully be pretty obvious to everyone.

Reply:
The reason is simply, that these types are only generic during type checking, but not during runtime, making CPython throw its hands up in the air during compilation and throwing a TypeError. `TYPE_CHECKING` is only true during type checking time and guards against this. I've added a comment.